perception/image_projection_based_fusion/include/image_projection_based_fusion/pointpainting_fusion/node.hpp

```diff
@@ -33,6 +33,13 @@ namespace image_projection_based_fusion
 {
 using Label = autoware_auto_perception_msgs::msg::ObjectClassification;
 
+inline bool isInsideBbox(
```

Comment:
zc means scaling, right?

- I prefer more specific name like `roi_expand_scale`or `scale_factor`
- Current calculation will not properly expand original rois (see figure below)

How about following expression?

```cpp
inline bool isInsideBbox(
  float proj_x, float proj_y, const sensor_msgs::msg::RegionOfInterest& roi, float scale_factor)
{
  // Calculate new width and height with scaling
  float new_width = roi.width * scale_factor;
  float new_height = roi.height * scale_factor;

  // Calculate new offsets to keep the center fixed
  float new_x_offset = roi.x_offset + (roi.width - new_width) / 2.0;
  float new_y_offset = roi.y_offset + (roi.height - new_height) / 2.0;

  // Check if the projected point is within the scaled ROI
  return proj_x >= new_x_offset && proj_x <= (new_x_offset + new_width) &&
         proj_y >= new_y_offset && proj_y <= (new_y_offset + new_height);
}
```

![image](https://github.com/autowarefoundation/autoware.universe/assets/20086766/6a81a1df-f173-4494-bb90-dfee664c71ba)

Reply:
zc means the z-axis value of the pointcloud in the camera coordinate, the formula is commented [here](https://github.com/autowarefoundation/autoware.universe/pull/7169/files#diff-e8df78707e8cee5e869c96c349aca84fa2126bc70e25f093bf10f8a325e638e2L314-L320). zc is multified to the roi offset to avoid division calculation.